libexec/basher-link

```diff
@@ -1,23 +1,53 @@
 #!/usr/bin/env bash
 # Summary: Installs a local directory as a basher package
-# Usage: basher link <directory>
+# Usage: basher link <directory> <package>
 
 set -e
 
-if [ "$#" -ne 1 ]; then
+if [ "$#" -ne 2 ]; then
   basher-help link
   exit 1
 fi
 
 directory="$1"
+package="$2"
 
 if [ ! -d "$directory" ]; then
   echo "Directory '$directory' not found."
   exit 1
 fi
 
-mkdir -p "${BASHER_PACKAGES_PATH}/link"
-ln -s "$(pwd)/$directory" "${BASHER_PACKAGES_PATH}/link/$directory"
+if [ -z "$package" ]; then
+  basher-help link
+  exit 1
+fi
+
+IFS=/ read -r namespace name <<< "$package"
+
+if [ -z "$namespace" ]; then
+  basher-help link
+  exit 1
+fi
+
+if [ -z "$name" ]; then
+  basher-help link
+  exit 1
+fi
+
+if [ -d "${BASHER_PACKAGES_PATH}/$package" ]; then
+  echo "Package '$package' is already present"
+  exit 0
```

Comment:
Not sure about the status code we should return here. Should it be considered an error if the package is already present?

Also, should we check if the existing package is
- a symlink to the given directory
- a symlink to another directory
- a directory

...to improve the error message?

Reply:
The most sane option would be to fail if the directory already exists.

Reply:
Done